### PR TITLE
Python NumPy - Math functions .nanprod()

### PR DESCRIPTION
**Pull request by gourabdg47**
Created new file docs/content/numpy/concepts/math-methods/terms/nanprod/nanprod.md

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [+] All writings are my own.
- [+] My entry follows the Codecademy Docs style guide.
- [+] My changes generate no new warnings.
- [+] I have performed a self-review of my own writing and code.
- [+] I have checked my entry and corrected any misspellings.
- [+] I have made corresponding changes to the documentation if needed.
- [+] I have confirmed my changes are not being pushed from my forked `main` branch.
- [+] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [+] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by gourabdg47** [comment]
Issue : 
https://github.com/Codecademy/docs/issues/5243

File: content/numpy/concepts/math-methods/terms/nanprod/nanprod.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  - 'Math'
+  - 'NumPy'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+  - 'paths/data-science'
+  - 'paths/data-science-foundations'
+---
+
+In NumPy, the **`.nanprod()`** function calculates the product of array elements over a specified axis, treating NaN (Not a Number) values as 1. This function is particularly useful when working with arrays that may contain missing or undefined values represented as NaN.
```
Comment (cigar-galaxy82) [inline content/numpy/concepts/math-methods/terms/nanprod/nanprod.md:19]:
```suggestion
In NumPy, the **`.nanprod()`** function calculates the product of array elements over a specified axis, treating `NaN` (Not a Number) values as 1. This function is particularly useful when working with arrays that may contain missing or undefined values represented as `NaN`.
```

File: content/numpy/concepts/math-methods/terms/nanprod/nanprod.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+```pseudo
+numpy.nanprod(a, axis=None, dtype=None, out=None, keepdims=False)
+```
+
+- `a`: The input array.
+- `axis` (optional): The axis along which the product is computed. If not specified, the product of all elements is calculated.
+- `dtype` (optional): The data type of the output. If not specified, the dtype of the input is used.
+- `out` (optional): An array to store the result. If not provided, a new array is created.
+- `keepdims` (optional): If set to True, the axes which are reduced are left in the result as dimensions with size one.
```
Comment (cigar-galaxy82) [inline content/numpy/concepts/math-methods/terms/nanprod/nanprod.md:31]:
```suggestion
- `a`: An array for which the product is calculated.
- `axis`: An optional parameter used as the axis along which the product is computed. If not specified, the product of all elements is calculated.
- `dtype` : An optional parameter used as the data type of the output. If not specified, then the data type the input is used.
- `out`: An optional parameter specifying the array to store the result. If not provided, a new array is created.
- `keepdims`: An optional parameter if set to True, the axes that are reduced are left in the result as dimensions with size one.
```

File: content/numpy/concepts/math-methods/terms/nanprod/nanprod.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## Example 1
+
+This example demonstrates using `.nanprod()` function to calculate the product of an array containing NaN values:
```
Comment (cigar-galaxy82) [inline content/numpy/concepts/math-methods/terms/nanprod/nanprod.md:35]:
```suggestion
This example demonstrates using `.nanprod()` function to calculate the product of an array containing `NaN` values:
```

File: content/numpy/concepts/math-methods/terms/nanprod/nanprod.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
+---
+Title: '.nanprod()'
+Description: 'Calculates the product of array elements over a given axis, treating NaN (Not a Number) values as 1.'
```
Comment (cigar-galaxy82) [inline content/numpy/concepts/math-methods/terms/nanprod/nanprod.md:3]:
```suggestion
Description: 'Calculates the product of array elements.'
```

File: content/numpy/concepts/math-methods/terms/nanprod/nanprod.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+result = np.nanprod(arr, axis=(0, 1), keepdims=True)
+
+print("Original array:")
+print(arr)
+print("\nResult:")
+print(result)
+```
+
+This example showcases how `.nanprod()` handles NaN values and maintains the dimensions of the output when `keepdims=True`.
```
Comment (cigar-galaxy82) [inline content/numpy/concepts/math-methods/terms/nanprod/nanprod.md:94]: Give a blank line after this

File: content/numpy/concepts/math-methods/terms/nanprod/nanprod.md (hunk 4 above)
Comment (cigar-galaxy82) [reply content/numpy/concepts/math-methods/terms/nanprod/nanprod.md:3]: reframed as this gives a basic description and later in-depth explanation is given

**Review by cigar-galaxy82 — CHANGES_REQUESTED** [review]

**Comment by cigar-galaxy82** [comment]
Hi @gourabdg47 thank you for contributing to Codecademy docs, I have suggested a few changes please modify them.

**Comment by gourabdg47** [comment]
Hi @cigar-galaxy82  I have committed your  suggestions

**Review by cigar-galaxy82 — APPROVED** [review]

**Review by cigar-galaxy82 — APPROVED** [review]

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing @gourabdg47 😄 

The entry is ready to be merged 🚀

**Pull request merged**